src/views/Referrals/CreateAffiliateCode.js

```diff
@@ -0,0 +1,173 @@
+            Connect Wallet
+          </button>
+        )}
+      </div>
+    </div>
+  );
+}
+
+export function CreateAffiliateCodeForm({
+  handleCreateReferralCode,
+  recentlyAddedCodes,
+  setRecentlyAddedCodes,
+  callAfterSuccess,
+}) {
+  const [referralCode, setReferralCode] = useState("");
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [error, setError] = useState("");
+  const inputRef = useRef("");
+  const [referralCodeCheckStatus, setReferralCodeCheckStatus] = useState("ok");
+  const debouncedReferralCode = useDebounce(referralCode, 300);
+  const { account, chainId } = useWeb3React();
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+
+  useEffect(() => {
+    let cancelled = false;
+    const checkCodeTakenStatus = async () => {
+      if (error || error.length > 0) {
+        setReferralCodeCheckStatus("ok");
+        return;
+      }
+      const { status: takenStatus } = await getReferralCodeTakenStatus(account, debouncedReferralCode, chainId);
+      // ignore the result if the referral code to check has changed
+      if (cancelled) {
+        return;
+      }
+      if (takenStatus === "none") {
+        setReferralCodeCheckStatus("ok");
+      } else {
+        setReferralCodeCheckStatus("taken");
+      }
+    };
+    setReferralCodeCheckStatus("checking");
+    checkCodeTakenStatus();
+    return () => {
+      cancelled = true;
+    };
+  }, [account, debouncedReferralCode, error, chainId]);
+
+  function getButtonError() {
+    if (!debouncedReferralCode || debouncedReferralCode.length === 0) {
+      return "Enter a code";
+    }
+    if (referralCodeCheckStatus === "taken") {
+      return "Code already taken";
+    }
+    if (referralCodeCheckStatus === "checking") {
+      return "Checking code...";
+    }
+
+    return false;
+  }
+
+  const buttonError = getButtonError();
+
+  function getPrimaryText() {
+    if (buttonError) {
+      return buttonError;
+    }
+
+    if (isProcessing) {
+      return `Creating...`;
+    }
+
+    return "Create";
+  }
+  function isPrimaryEnabled() {
+    if (buttonError) {
+      return false;
+    }
+    if (error || isProcessing) {
+      return false;
+    }
+    return true;
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    setIsProcessing(true);
+    const { status: takenStatus, info: takenInfo } = await getReferralCodeTakenStatus(account, referralCode, chainId);
+    if (takenStatus === "all" || takenStatus === "current") {
+      setIsProcessing(false);
+    }
+    if (takenStatus === "other") {
+      setIsProcessing(false);
+    }
+
+    if (takenStatus === "none" || takenStatus === "other") {
+      const ownerOnOtherNetwork = takenInfo[chainId === ARBITRUM ? "ownerAvax" : "ownerArbitrum"];
+      try {
+        const tx = await handleCreateReferralCode(referralCode);
+        if (typeof callAfterSuccess === "function") {
```

Comment:
`callAfterSuccess` should be either undefined or function? i think it better to just check `if (callAfterSuccess) ...`

in case parent function passed callAfterSuccess of incorrect type it's better to throw an error, and parent function code will be fixed

Reply:
and inside [JoinReferralCode](https://github.com/gmx-io/gmx-interface/pull/141/files#diff-4d09bd609cef3a6437b9da1f2d741182d917e13b4c104342dd0d771a09b7ac25)